packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -585,6 +599,10 @@ export const MenuBarMixin = (superClass) =>
      * @protected
      */
     _onMouseOver(e) {
+      if (this.openOnHover) {
+        this._preventClose = true;
+      }
```

Comment:
All the tests pass with this block removed. Needs a test where an opened menu bar doesn't close when the menu bar buttons are hovered over.